monkey/monkey_island/cc/services/reporting/stolen_credentials.py

```diff
@@ -0,0 +1,59 @@
+import logging
+from typing import Mapping, Sequence
+
+from common.common_consts.credential_component_type import CredentialComponentType
+from monkey_island.cc.models import StolenCredentials
+
+logger = logging.getLogger(__name__)
+
+
+def get_stolen_creds() -> Sequence[Mapping]:
+    stolen_creds = _fetch_from_db()
+    stolen_creds = _format_creds_for_reporting(stolen_creds)
+
+    logger.info("Stolen creds generated for reporting")
+    return stolen_creds
+
+
+def extract_ssh_keys(credentials: Sequence[Mapping]) -> Sequence[Mapping]:
+    ssh_keys = []
+    for credential in credentials:
+        if credential["_type"] == CredentialComponentType.SSH_KEYPAIR.name:
+            ssh_keys.append(credential)
+    return ssh_keys
```

Comment:
Consider using a list comprehension:

```suggestion
    return [c for c in credentials if c["_type"] == CredentialComponentType.SSH_KEYPAIR.name]
```